### ISSUE TITLE
Document ENV vars available to builds

### ISSUE THREAD

**Issue by faceleg** [lambci/lambci#37]
Copy from:  

 - https://github.com/lambci/lambci/blob/master/utils/config.js#L102

**Comment by afaqurk** [comment]
Good find. 

@mhart Are these environment variables intended to be "private" to lambci internal code? Do you see any issue with documenting them for usage by the builds themselves?

**Comment by mhart** [comment]
@afaqurk no, not at all – all of the variables there can and should be used by builds if necessary!

Just a matter of adding it to the documentation somewhere

**Comment by afaqurk** [comment]
I just tried putting them in a new [Environment Variables](https://github.com/afaqurk/lambci#environment-variables) section below [Default Configuration](https://github.com/afaqurk/lambci#default-configuration) section.

The idea being that anywhere the documentation needs to remind of or mention the default environment variables, they can just link to it.
 
What do you think?

**Comment by afaqurk** [comment]
Alternatively, the simpler approach could be documenting them inline with the code via comments and just pointing the docs to the line @faceleg mentioned.

```js
exports.prepareBuildConfig = function(build) {
  var buildConfig = {
    env: {
      CI: true, // Signifies that this a build triggered by a CI pipeline.
      LAMBCI: true, // Signifies that lambCI triggered this build.
      LAMBCI_REPO: build.repo, // URL for the repository for the target source code for this build
      LAMBCI_BRANCH: build.branch, // Branch name for the git repository to build the source from.
     ...
```

**Comment by mhart** [comment]
@afaqurk your example looks great IMO – would totally accept a PR like that – thanks for the effort!

**Comment by faceleg** [comment]
That looks awesome @afaqurk. My humble suggestion is to also provide a link to the source file, as some people might find that easier to read (myself)
